core/templates/dev/head/pages/admin/Admin.js

```diff
@@ -46,8 +46,8 @@ oppia.controller('Admin', [
       $scope.statusMessage = statusMessage;
     };
 
-    $scope.$watch(function() {
-      return $window.location.hash;
-    }, AdminRouterService.showTab);
+    $scope.$on('$locationChangeSuccess', function() {
+      AdminRouterService.showTab(location.hash.replace('/', ''));
```

Comment:
Is this supposed to be `$location`? If not, where is `$location` used?

Reply:
$locationChangeSuccess is an event of $location which injected in order for the scope to be able to subscribe to this event

Reply:
So, is it the case that, if you remove $location from the dependency list, this does not work?

Reply:
yes

Reply:
Ah, ok. I think you might want to take @AllanYangZhou's (implicit) suggestion to replace `location` with `$location`, assuming that works, otherwise this might trigger an "unused dependency" error in our linter in the future.

Reply:
I tried but it didn't work :(

Reply:
You might need to make it a function call, see the docs. https://docs.angularjs.org/api/ng/service/$location#hash

Reply:
Tried this:
```
$scope.$watch(function() {
      return $location.hash();
    }, function() {
      AdminRouterService.showTab(location.hash.replace('/', ''));
    });
```
Doesn't work

Reply:
Ah, I see. Yep, sounds good -- whatever works!